src/R/Components/Impl/ConnectionManager/Implementation/View/ConnectionManagerControl.xaml

```diff
@@ -126,116 +125,65 @@
+                        <!-- Name, path, command line  -->
                         <TextBlock Grid.Column="2" Margin="0,1,6,2" Text="{Binding Path=Name}"/>
-                        <Separator Grid.Column="3" Margin="0,1,6,2" Style="{StaticResource VerticalSeparator}"/>
+                        <Separator Grid.Column="3" Margin="0,1,6,2" Style="{StaticResource VerticalSeparator}"
+                                   Visibility="{Binding Path=IsRemote, Converter={x:Static rwpf:Converters.FalseIsCollapsed}}"/>
                         <TextBlock Grid.Column="4" Margin="0,1,3,2" Text="{Binding Path=Path, Mode=OneWay}"
-                                   TextTrimming="CharacterEllipsis" TextWrapping="NoWrap"/>
+                                   TextTrimming="CharacterEllipsis" TextWrapping="NoWrap"
+                                   Visibility="{Binding Path=IsRemote, Converter={x:Static rwpf:Converters.FalseIsCollapsed}}"/>
+
+                        <!-- connect button -->
+                        <Button x:Name="ButtonConnect" Grid.Column="5" Style="{StaticResource ToolWindowButtonStyle}" Click="ButtonConnect_Click" 
+                                Visibility="{Binding Path=IsConnected, Converter={x:Static rwpf:Converters.TrueIsCollapsed}}"
+                                Padding="0,0,8,0">
+                            <Button.ToolTip>
+                                <TextBlock Text="{Binding Path=Name, StringFormat={x:Static components:Resources.ConnectionManager_ConnectTooltip}}" />
+                            </Button.ToolTip>
+                            <Image Source="{x:Static rwpf:ImageSources.ConnectArrow}"/>
+                        </Button>
 
-                        <!-- edit button -->
-                        <Button x:Name="ButtonEdit" Grid.Column="5" Style="{StaticResource ToolWindowButtonStyle}" Click="ButtonEdit_Click"
-                                ToolTip="{x:Static components:Resources.ConnectionManager_EditTooltip}" Visibility="Collapsed">
-                            <Image Source="{x:Static rwpf:ImageSources.Edit}" />
+                        <!-- edit/properties button -->
+                        <Button x:Name="ButtonEdit" Grid.Column="6" Style="{StaticResource ToolWindowButtonStyle}" Click="ButtonEdit_Click"
+                                ToolTip="{x:Static components:Resources.ConnectionManager_EditTooltip}">
+                            <Image Source="{x:Static rwpf:ImageSources.Property}"/>
                         </Button>
 
                         <!-- delete button -->
-                        <Button x:Name="ButtonDelete" Grid.Column="6" Style="{StaticResource ToolWindowButtonStyle}" Click="ButtonDelete_Click"
-                                ToolTip="{x:Static components:Resources.ConnectionManager_DeleteTooltip}" Visibility="Collapsed">
+                        <Button x:Name="ButtonDelete" Grid.Column="7" Style="{StaticResource ToolWindowButtonStyle}" Click="ButtonDelete_Click"
```

Comment:
I considered it, but UI looks a bit odd with gaps on some entries. I'd like to gather some feedback.